app/Http/Controllers/V2/AuditStatus/GetAuditStatusController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Http\Controllers\V2\AuditStatus;
+
+use App\Http\Controllers\Controller;
+use App\Http\Resources\V2\AuditStatusResource;
+use App\Models\V2\AuditStatus\AuditStatus;
+use App\Models\V2\Projects\Project;
+use App\Models\V2\Sites\Site;
+use App\Models\V2\Sites\SitePolygon;
+use Illuminate\Http\Request;
+
+class GetAuditStatusController extends Controller
+{
+    public function __invoke(Request $request, string $id = null)
+    {
+        $auditStatus = AuditStatus::where('entity', $request->input('entity'))
+            ->where('entity_uuid', $request->input('uuid'))
+            ->orderBy('updated_at', 'desc')
+            ->orderBy('created_at', 'desc')
+            ->with('auditAttachments')
+            ->get();
+
+        $audit_statuses_with_entity = $this->mapAuditStatus($auditStatus);
+
+        return AuditStatusResource::collection($audit_statuses_with_entity);
+    }
+
+    private function mapAuditStatus($auditStatus)
+    {
+        return $auditStatus->map(function ($audit) {
+            return [
+                'id' => $audit->id,
+                'entity' => $this->getEntity($audit->entity, $audit->entity_uuid)->name,
+                'entity_uuid' => $audit->entity_uuid,
+                'status' => $audit->status,
+                'comment' => $audit->comment,
+                'date_created' => $audit->date_created,
+                'created_by' => $audit->created_by,
+                'type' => $audit->type,
+                'is_submitted' => $audit->is_submitted,
+                'is_active' => $audit->is_active,
+                'first_name' => $audit->first_name,
+                'last_name' => $audit->last_name,
+                'request_removed' => $audit->request_removed,
+                'attachments' => $this->mapAttachments($audit->auditAttachments),
```

Comment:
This work should all be done in `AuditStatusResource` in the `toArray` method, that's the purpose of that method in a resource. With the changes mentioned above, the the `'entity'` mapping would use (in the resource, `$this` maps to the underlying model being mapped): `$this->auditable->name`, and the association should already be mapping memory instead of requiring another query - `->auditable` will come from the `morphTo` association.